haystack/document_stores/types/filter_policy.py

```diff
@@ -52,10 +305,27 @@ def apply_filter_policy(
           values from the runtime filters will overwrite those from the initial filters.
     :param init_filters: The initial filters set during the initialization of the relevant retriever.
     :param runtime_filters: The filters provided at runtime, usually during a query operation execution. These filters
-                            can change for each query/retreiver run invocation.
+                            can change for each query/retriever run invocation.
+    :param logical_operator: The default logical operator to use when merging filters (non-legacy filters only).
```

Comment:
If it's meant to be use as the default operator, let name it `default_logical_operator`.